### PR TITLE
Comparison tool shareable URL generator 

### LINKED ISSUE #844: [Feature Request] comparison tool URL generator
When we come up with a comparison of pbench results, it would be nice to share the comparison, but not in static form (pdf).  It would be very useful to share a link, that when loaded, would re-generate the comparison.  Right now, when generating a comparison, the URL in the browser is not unique, so we simply cannot copy that link today.  What we would like is a button to generate a "sharable link", with enough information embedded in the link to regenerate the same comparison by anyone who received the link.

### PR DESCRIPTION
**Pull request by Anisha1234**
Will fix #844 

In order to enable productivity features within pbench dashboard and eventually benchpress dashboard in the future — our platform requires an efficient backend for supporting dynamic storage and retrieval of user data. 
This platform consists of a Postgres database exposed via a GraphQL API server. This API server is powered by Prisma in order to enable high performance querying, a GUI client for visualizing queries, and a realtime event system for database event tracking.

The backend environment is currently being tracked at the following GitHub repo:
https://github.com/gurbirkalsi/pbench_dashboard_backend

This PR involves:

- UI for shareable URL generation
- Saving whole redux store data to Postgres database
- Retrival and Re-rendering of entire redux component with the fetched data.

### CODE REVIEW

**Comment by portante** [comment]
@Anisha1234, we removed the "`[WIP]`" text from the title of this PR since we have a "`WIP`" label we can use instead.

When you feel this PR is ready to be removed, can you remove the "`WIP`" label, and request review?  In the mean time, can you consider rebasing this work and squashing the commits down to one?

We have moved this PR to the 0.60 release since we are getting ready to push 0.59 out the door shortly.

**Comment by gurbirkalsi** [comment]
@Anisha1234 I've included a solution to this PR that utilizes the Web Storage API to modify the persisted information stored in localStorage. After we query the shared config from the backend, we can rehydrate the redux session by replacing the persisted information in localStorage and then redirect the user the pathname designated in the shared config.

**Comment by gurbirkalsi** [comment]
@Anisha1234 Can we also consider creating a `ShareModal` component for better organization in the `GlobalHeader` page component? I've included a CodeSandbox snippet here for a simplified UX design pattern we can follow to manage state and handle confirmation modals as a result of the callback from link generation.

File: web-server/v0.4/src/components/SessionModal/index.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    this.setState({
+      visible: true,
+    });
+  };
+
+  onGenerate = () => {
+    const { configData } = this.props;
+    const { description } = this.state;
+    const stringProp = JSON.stringify(configData);
+    axios({
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/components/SessionModal/index.js]: Can we move this query to the GraphQL API to `services/global.js` and reference `dispatch` as a prop passed down from the `GlobalHeader` page component?

File: web-server/v0.4/src/components/SessionModal/index.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      url: 'http://localhost:4466/',
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/components/SessionModal/index.js]: All endpoints for a query should be referenced from the `datastoreConfig` prop as part of the global redux store. We can define the endpoint within `mock/datastoreConfig.js` as an additional key called `backend`.

File: web-server/v0.4/src/components/SessionModal/index.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    }).then(result => {
+      this.setState({
+        confirmLoading: true,
+      });
+      if (document.getElementById('description')) {
+        setTimeout(() => {
+          this.setState({
+            visible: false,
+            confirmLoading: false,
+            description: document.getElementById('description').value,
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/components/SessionModal/index.js]: Can we update the description using the ant.design API that offers methods such as `onChange`? This way we can avoid referencing the element ID.

File: web-server/v0.4/src/components/SessionModal/index.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+              }
+            }       
+          `,
+        variables: {
+          config: stringProp,
+          description,
+        },
+      },
+    }).then(result => {
+      this.setState({
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/components/SessionModal/index.js]: Can we handle loading via the global service you create for this query? We follow a similar pattern in other page components such as `Controllers`: https://github.com/distributed-system-analysis/pbench/blob/4b252d0848f69eae55bd4060c872b12b756c2944/web-server/v0.4/src/pages/Controllers/index.js#L17-L20

File: web-server/v0.4/src/components/SessionModal/index.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    message.success(`Copied the link: ${generatedUrl.value}`);
+  };
+
+  render() {
+    const { styles } = this.props;
+    const { visible, confirmLoading, description } = this.state;
+    return (
+      <span>
+        <Tooltip title="Share" onClick={this.showModal}>
+          <a className={styles.action}>
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/components/SessionModal/index.js]: We reference `styles.action` here and define `styles` as a required prop; however we don't have an entry for `action` defined within index.less. Is there a reason we are referencing this and requiring it as a prop?

File: web-server/v0.4/src/pages/RunComparison/index.js [hunk 6]
```diff
@@ -101,12 +101,13 @@ class RunComparison extends React.Component {
       exportModalVisible: false,
       defaultComponents: ['details', 'summary', 'table', 'timeseries'],
       selectedComponents: [],
-      pdfHeader: '',
-      pdfName: '',
+      pdfHeader: ' ',
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/pages/RunComparison/index.js]: A few unrelated changes seem to have leaked into this PR - can we clean this up?

File: web-server/v0.4/src/components/SessionModal/index.js (hunk 3 above)
Comment (manaswinidas) [reply web-server/v0.4/src/components/SessionModal/index.js]: @gurbirkalsi Sure

Comment (Anisha1234) [reply web-server/v0.4/src/components/SessionModal/index.js]: Sure

File: web-server/v0.4/src/components/SessionModal/index.js (hunk 5 above)
Comment (Anisha1234) [reply web-server/v0.4/src/components/SessionModal/index.js]: In session modal, we need to reference the style pattern of the globalHeader. In order to import the style, I referenced it as a prop.

File: web-server/v0.4/src/components/SessionModal/index.js (hunk 2 above)
Comment (gurbirkalsi) [reply web-server/v0.4/src/components/SessionModal/index.js]: @Anisha1234 This item still requires attention as we would like to reference the production endpoint for the GraphQL server via `datastoreConfig`

File: web-server/v0.4/src/services/dashboard.js [hunk 7]
```diff
@@ -117,6 +117,24 @@ export async function queryTocResult(params) {
   return request.post(endpoint);
 }
 
+export async function querySharedConfig(params) {
+  const { id } = params;
+  return request.post('http://localhost:4466/',
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/services/dashboard.js]: We should reference the `graphql` key from our datastoreConfig field within the Redux store in order to make requests against the production GraphQL service.

File: web-server/v0.4/src/pages/RunComparison/index.js [hunk 8]
```diff
@@ -277,7 +277,7 @@ class RunComparison extends React.Component {
     const expandedRowRender = cluster => {
       const expandedColumns = [
         {
-          title: 'iteration_name',
+          title: 'iteration_names',
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/pages/RunComparison/index.js]: It seems an unrelated change worked its way into this PR.

**Comment by gurbirkalsi** [comment]
@Anisha1234 It seems our change in PR #1271 for creating local storage keys used by redux-persist is breaking the redirect workflow with url generation. I'll submit a fix independent of this PR that we can track and reference here before we merge this work.

**Comment by portante** [comment]
@Anisha1234, so where are we at with this PR?

Can you rebase and address the comments from passed reviews, resolving those comments once you address them?

Thanks!

**Comment by Anisha1234** [comment]
> @Anisha1234 Can we also consider creating a `ShareModal` component for better organization in the `GlobalHeader` page component? I've included a CodeSandbox snippet here for a simplified UX design pattern we can follow to manage state and handle confirmation modals as a result of the callback from link generation: https://codesandbox.io/s/0qqjzv7o7l

Implemented.

**Comment by Anisha1234** [comment]
@portante I believe, as per my last discussion with @gurbirkalsi , we were going to create a new PR to track this work and start the review process. As per my knowledge, this PR was working in both my and Gurbir's system.

**Comment by gurbirkalsi** [comment]
Closing in favor of PR #1435
